hack/vendor-console.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash 
+# 
+# This script vendors the Web Console source files into bindata.go to allow ....
+#
+#  Accepted environment variables are:
+#   - GIT_REF:           specifies which branch / tag of the web console to vendor. If set then untracked/uncommitted changes
+#                        will be stashed. If not set then the current working state of the web console directory will be used.
+#   - CONSOLE_REPO_PATH: specifies a directory path to look for the web console repo.  If unset it is assumed to be
+#                        a sibling to this repository.
+
+set -o errexit 
+set -o nounset 
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/common.sh"
+
+CONSOLE_REPO_PATH=${CONSOLE_REPO_PATH:-$OS_ROOT/../origin-web-console}
+if [[ ! -d "$CONSOLE_REPO_PATH" ]]; then
+  echo "The console repo at path $CONSOLE_REPO_PATH does not exist."
+  echo "Make sure you have cloned the web console repo locally:  git@github.com:openshift/origin-web-console.git"
+  echo "Or, you can specify a different path with the CONSOLE_REPO_PATH environment variable."
+  exit 1
+fi
+
+echo "Making sure go-bindata binary has been built..."
+pushd "${OS_ROOT}/Godeps/_workspace" > /dev/null
+  godep_path=$(pwd)
+  pushd src/github.com/jteeuwen/go-bindata > /dev/null
+    GOPATH=$godep_path go install ./...
+  popd > /dev/null
+popd > /dev/null
+
+if [[ -z "${GIT_REF:+x}" ]]; then 
+  echo "No git ref specified, using current state of the repo including any unstaged changes...";
+else 
+  echo "Using git ref ${GIT_REF}, all untracked and uncommitted changes will be stashed..."
+  # TODO improve this to return the user back to the state they were in
```

Comment:
I think a cleaner approach with fewer assumptions on how people want to use `git` would be to detect a dirty workspace, tell the user to fix it, and fail instead of taking on the job of stashing, etc.